ocaml/xapi/cli_operations.ml

```diff
@@ -2690,6 +2690,7 @@ let snapshot_copy printer rpc session_id params =
 let snapshot_destroy printer rpc session_id params =
   let snap_uuid = get_snapshot_uuid params in
   let snap_ref = Client.VM.get_by_uuid rpc session_id snap_uuid in
+  if (not Client.VM.get_is_a_snapshot rpc session_id snap_ref) then failwith "This operation can only destroy VM snapshot.";
```

Comment:
The whole expression `Client.VM.get_is_a_snapshot rpc session_id snap_ref` is of type bool, so you need to wrap it in parentheses before applying `not`
i.e. `(not (Client.VM.get_is_a_snapshot rpc session_id snap_ref))`

Reply:
Not quite. It's enough to do 
```
if not (Client.VM.get_is_a_snapshot rpc session_id snap_ref) then ...
```

You can avoid binding names that are only used once (but this is a matter of taste). I'd also use `vm` or `snapshot` as a name as I don't think types should be encoded into names.

```
let vm = get_snapshot_uuid params |>  Client.VM.get_by_uuid rpc session_id in
```

Reply:
Thanks for catching this out! I made the change to parenthesize what's after not when I build and test, but seems missed the updates when I do PR.